### PR TITLE
Fix hardcoded FEATURE_DISTRO_AGNOSTIC_SSL

### PR DESCRIPTION
**Pull request by omajid**
This is a fix for https://github.com/dotnet/runtime/issues/41768

### CODE REVIEW

**Comment by omajid** [comment]
cc @tmds

**Comment by dagood** [comment]
macOS error is a homebrew issue, fixed here: https://github.com/Homebrew/homebrew-cask-versions/pull/9633.

**Comment by omajid** [comment]
I just realized that this doesn't handle the case where `FEATURE_DISTRO_AGNOSTIC_SSL` is defined... because it's never defined in the installer configuration, only in libraries :cry:

**Comment by omajid** [comment]
dotnet/runtime PR: https://github.com/dotnet/runtime/pull/42415

**Comment by omajid** [comment]
The dotnet/runtime PR was merged, into `master`. That means this wont be a part of 5.0 GA, right? What's the best thing to do from a source-build point of view now? Should we get this PR merged or do something else?

**Comment by dagood** [comment]
> That means this wont be a part of 5.0 GA, right?

Yeah, my understanding is `master` is solidly 6.0 at this point, no automatic backporting.

> What's the best thing to do from a source-build point of view now? Should we get this PR merged or do something else?

I think that we should get this patch in to make sure it keeps building correctly in dotnet/source-build up to 5.0 GA no matter what. Then we can remove it naturally once it gets ported to 5.0 (whether that's before or after GA).

I think there's just the one nit from the dotnet/runtime PR to bring back here, the message. I think we should put a link to the dotnet/runtime PR in the patch description to make it easy to navigate later. (E.g. if it's auto-applied, show up nicely in the commit message.) Then I think it's good to go.

@crummel @dseefeld

File: patches/runtime/0014-Fix-singlefilehost-build-in-non-portable-mode-42415.patch [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
++    if(NOT FEATURE_DISTRO_AGNOSTIC_SSL)
++        find_package(OpenSSL)
++    endif()
++
+     if(CLR_CMAKE_TARGET_LINUX OR CLR_CMAKE_TARGET_FREEBSD OR CLR_CMAKE_TARGET_NETBSD OR CLR_CMAKE_TARGET_SUNOS)
+         # These options are used to force every object to be included even if it's unused.
+         set(START_WHOLE_ARCHIVE -Wl,--whole-archive)
+@@ -212,3 +217,10 @@ target_link_libraries(singlefilehost
+     ${NATIVE_LIBS}
+     ${END_WHOLE_ARCHIVE}
```
Comment (omajid) [inline patches/runtime/0014-Fix-singlefilehost-build-in-non-portable-mode-42415.patch:49]: I had to manually massage these few lines for the patch to apply.

**Review by dagood — APPROVED** [review]

**Review by crummel — APPROVED** [review]

**Review by dseefeld — APPROVED** [review]

**Pull request merged**